File: matlab/toolbox_wavelet_meshes/toolbox/clamp.m
function y = clamp(x,a,b)

% clamp - clamp a value
%
%   y = clamp(x,a,b);
%
% Default is [a,b]=[0,1].
%
%   Copyright (c) 2004 Gabriel Peyré

if nargin<2
    a = 0;
end
if nargin<3
    b = 1;
end

y = max(x,a);
y = min(y,b);

File: r/nt_toolbox/toolbox_general/clamp.R





clamp <- function(x, a=0, b=1){
  ####
  # clamp - clamp a value
  #
  #   y = clamp(x,a,b);
  #
  # Default is [a,b]=[0,1].
  ####
  x[ x>b ] <- b
  x[ x<a ] <- a
  return(x)
}

